third_party/terraform/resources/resource_dataflow_job.go

```diff
@@ -96,6 +96,18 @@ func resourceDataflowJob() *schema.Resource {
 				Optional: true,
 				ForceNew: true,
 			},
+
+			"network": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
```

Comment:
Generally on "resource reference" fields, GCP will accept a short name (`default`, inferring project and location based on the parent resource), partial uri  (`projects/my-project/global/networks/default`) or a full self link (`https://www.googleapis.com/compute/v1/projects/my-project/global/networks/default` or something). It will then (generally) return the full self link, leading to a diff in Terraform. It can also interact weirdly with the version; sending a `v1` self link to a `v1beta` API could return the string canonicalised as `v1beta`, or untouched as `v1`.

We use a common utility function called a diff suppress func to deal with this; if this fn returns `true`, Terraform considers the before and after values identical even if they're distinct strings. Here's an example of the one we use for self links, `compareSelfLinkOrResourceName`: https://github.com/GoogleCloudPlatform/magic-modules/blob/master/third_party/terraform/resources/resource_compute_instance.go#L147